usersimcrs/items/ratings.py

```diff
@@ -125,13 +141,31 @@ def create_split(
             historical_ratio: Ratio ([0..1]) of ratings to be used as historical
               data.
 
+        Raises:
+            ValueError: if historical_ratio is not in the interval [0,1].
+
         Returns:
             Two Ratings objects, one corresponding to historical and another to
             ground truth ratings.
         """
+        if historical_ratio > 1.0 or historical_ratio < 0.0:
+            raise ValueError("historical_ratio is bounded in [0,1]")
+
+        # Determine the number of items to use as historical data.
+        nb_historical_items = int(historical_ratio * len(self._item_ratings))
+
         historical_ratings = Ratings(self._item_collection)
         ground_truth_ratings = Ratings(self._item_collection)
-        # TODO: Implement this method with tests
-        # See: https://github.com/iai-group/UserSimCRS/issues/108
+
+        for i, (item_id, user_ratings) in enumerate(self._item_ratings.items()):
```

Comment:
Here, we need to _randomly_ sample x% of the ratings _per user_ as historical data. For testability, it'd be good for this method to take a user_item_sampler Callable argument. (That way, we could have another sampler later if we want the first x% ratings based on timestamp as historical data, as opposed to random x%.)

Reply:
I am not entirely sure how you see the implementation of this sampler. I made a suggestion that is open for open discussion.